### ISSUE TITLE
Issue in Setting up Azure Dataflow using Common Data Model

### ISSUE THREAD

**Issue by nisargdshah** [microsoft/CDM#215]
I am currently exploring Common Data Model (CDM) in Azure as one of the PoCs I am doing for my company. I was able to successfully create a data flow in an ADF pipeline using model.json file. However, when I am trying to setup data flow using manifest file I am facing an issue as attached in the word document.

JSON file details are as attached. I am giving you screenshots in a word file (refer attachment Mapping DataFlow Setting Issue using Common Data Model.docx) as I am not able to validate the data flow in ADF author screen because getting error while importing the CDM schema.

Can anyone please help me out in resolving this issue.

Thanks & Regards,
Nisarg Shah
[Mapping DataFlow Setting Issue using Common Data Model.docx](https://github.com/microsoft/CDM/files/5842168/Mapping.DataFlow.Setting.Issue.using.Common.Data.Model.docx)

**Comment by Venkata1920** [comment]
Hi Nisarg Shah,

There seems to be an issue with the manifest file. The "location" should be inside a "datapartitions" array. Please refer https://docs.microsoft.com/en-us/common-data-model/cdm-manifest#data-partition-patterns for reference.

**Comment by nisargdshah** [comment]
Hi,

Thank you so much for rectifying the issue in the manifest file. Now I am able to import the schema. However, while previewing the data I am getting message as "No Output Data". Please find attached detailed document on this issue.

[Mapping DataFlow Setting Issue using Common Data Model_1.docx](https://github.com/microsoft/CDM/files/5846892/Mapping.DataFlow.Setting.Issue.using.Common.Data.Model_1.docx)

Can anyone please help me out in resolving this no output data issue.

Thanks & Regards,
Nisarg Shah

**Comment by Venkata1920** [comment]
Hi Nisarg Shah,

The "location" path is supposed to be the corpus path to the data file location. Based on your screenshot, it looks like the CSV file, the entity definition and the manifest file are in the same folder which is typically not advised. In any case, please specify the location with no path information just like you'd specify a file under the same folder. A better idea, though, would be to move the CSV file under a subfolder named "dataFiles" or similar and specify "location": "datafiles/person001.csv" .

**Comment by nisargdshah** [comment]
Hi Venkat,

Thanks for the solution. I tried what you have advised to put datafiles in separate folder and then giving the path in the location and it worked. I also tried with multiple files with varying filenames like product001.csv product002.csv and used below details in the manifest file and that also worked.

One more query to you is that do you find any better solution than the one I have mentioned below. Also if the filenames are appended with datetime format say "productddmmyyyyhhmiss.csv" e.g. "product21012021124325.csv" then what format we need to use in the datapartitionpattern array so that when the same code runs each day it should pick only that day's files keeping in mind that the datafile folder will be storing everyday file with filename format as mentioned above.

{
    "jsonSchemaSemanticVersion": "1.0.0",
    "imports": [{
        "corpusPath": "cdm:/foundations.cdm.json"
    }],
    "manifestName": "sales",
    "explanation": "CDM standard entities",
    "entities": [
        {
            "type": "LocalEntity",
            "entityName": "product",
            "entityPath": "product.cdm.json/product",
            "dataPartitions": [{
                                "location": "datafiles/product*.csv"
                               }
							  ],
            "dataPartitionPatterns": [{
            "name": "product",
            "rootLocation": "datafiles/", 
            "regularExpression": "(product(\\d+)\\.csv$)",
            "parameters": [
                "filenumber"
            ]
        }]
        }
    ]
}
Thanks & Regards,
Nisarg Shah

**Comment by Venkata1920** [comment]
Hi Nisarg Shah,

Your requirement is to be able to look up a dynamic value (today's date) which cannot be supported with regular expressions. As for the suggestion about the pattern, you could just use one of dataPartitions or dataPartitionPatterns. Both are not required. Given that the issue has been resolved, I'll close this one out.

**Comment by nisargdshah** [comment]
Hi Venkata,

Thanks for your inputs.

Is there any other way with which looking up dynamic value can be supported using CDM so that only required data can be loaded? Also, I did a scenario below

1. First file product001.csv had 10 records
2. Second file product002.csv had 2 records
3. Third file product003.csv had one record but this record was an update to one of the records in the first file. To give example in the first file record details were as
Product ID Product Name Product Category ID
10              P10                   2
In the third file I kept the values of above P10 record as seen in bold below.
Product ID Product Name Product Category ID
10              **Product 10       3**
So when I did data preview on it instead of taking the record in the third file as an UPDATE it took it as INSERT and showed me two records for the same product ID. So how would any update or deletion of records be handled in CDM?

Thanks & Regards,
Nisarg Shah
